### DISCUSSION TITLE
noOfLines feature - how to detect?

### DISCUSSION THREAD

**Discussion by adirgil** [chakra-ui/chakra-ui#7336] [Q&A]
let say i have - 
 `<Text noOfLines={3}>`
       ` Lorem Ipsum is simply dummy text of the printing and typesetting
        industry. Lorem Ipsum has been the industry's standard dummy text ever
        since the 1500s, when an unknown printer took a galley of type and`
     ` </Text>`
`{isTruncatedText && <Button>Read more</Button> }`

 `<Text noOfLines={3}>
        Lorem Ipsum is simply dummy text of the printing and typesetting.
      </Text>`
`{isTruncatedText  && <Button>Read more</Button> }`

there is a way to detect if the text reach the 3 lines and the text actually is truncated?
lets say i want to add a button that says "read more" and then set the "noOfLines" to undefined and show the all message.
there is a way that i can decide if to put this button or not? i want to show the button only when i indeed reach the all 3 lines.

the first one should have the button and the second one should not

**Comment by roitalpaz** [comment]
any solutions for this?

**Reply by vitalijalbu** [reply]
i used a state

**Comment by vitalijalbu** [comment]
@adirgil @roitalpaz 

```
const [isCollapsed, setIsCollapsed] = useState(true);
<Box>
  <Text noOfLines={isCollapsed ? 5 : undefined} overflow="hidden">
    {data?.description}
  </Text>
  <Button variant="link" colorScheme="primary" size="sm" onClick={() => setIsCollapsed(!isCollapsed)} mt={2}>
    {`read ${isCollapsed ? "more" : "less"}`}
  </Button>
</Box>

```

**Reply by jeanbaptistedeborman** [reply]
Seems insufficient as OP request is:  'more' should only be visible if isCollapsed AND text is truncated (text length > noOfLines)
